src/js/components/leaflet-map-marker.tsx

```diff
@@ -29,7 +29,7 @@ export class LeafletMapMarker extends React.Component<
       basestation.data.length > dataStore.frameNumber
     ) {
       actualTemp = basestation.data[dataStore.frameNumber].value;
-      predictedTemp = basestation.data[0].value;
+      predictedTemp = dataStore.predictionFor(basestation.id).temp;
```

Comment:
As mentioned below, I think `predictionFor()` should return `null` if there is no prediction, and so this code should check for a valid prediction before dereferencing it.